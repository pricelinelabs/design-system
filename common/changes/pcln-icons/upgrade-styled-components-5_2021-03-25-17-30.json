{
  "changes": [
    {
      "comment": "Upgrade to Styled Components 5 as devDependency",
      "type": "patch",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "craig.palermo@priceline.com"
}