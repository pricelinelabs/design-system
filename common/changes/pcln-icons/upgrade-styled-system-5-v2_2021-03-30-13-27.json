{
  "changes": [
    {
      "comment": "Upgrade to Styled System 5",
      "type": "major",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "craig.palermo@priceline.com"
}