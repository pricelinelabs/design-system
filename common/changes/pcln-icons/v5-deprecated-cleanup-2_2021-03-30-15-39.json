{
  "changes": [
    {
      "comment": "Remove deprecated props",
      "type": "major",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "craig.palermo@priceline.com"
}