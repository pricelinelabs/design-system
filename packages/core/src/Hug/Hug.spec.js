import React from 'react'
import { ThumbsUp } from 'pcln-icons'

import { Hug, Text, Card, theme } from '..'

const text = (
  <Text.span>
    Have you seen this <Text.span bold>card</Text.span> I‘m hugging?!
  </Text.span>
)

describe('Hug', () => {
  test('renders with border-radius from theme on top only', () => {
    const json = rendererCreateWithTheme(<Hug />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('border-top-left-radius', theme.radius)
    expect(json).toHaveStyleRule('border-top-right-radius', theme.radius)
    expect(json).not.toHaveStyleRule('border-bottom-left-radius', theme.radius)
    expect(json).not.toHaveStyleRule('border-bottom-right-radius', theme.radius)
  })

  test('renders text, icon, and Child', () => {
    const json = rendererCreateWithTheme(
      <Hug icon={<ThumbsUp />} text={text}>
        <Card p={3} color='text'>
          I‘m a card within a hug!
        </Card>
      </Hug>
    ).toJSON
    expect(json).toMatchSnapshot()
  })

  test('renders text when string is provided', () => {
    const json = rendererCreateWithTheme(
      <Hug
        icon={<ThumbsUp />}
        text="I've been known to hug a card from time to time"
      >
        <Card p={3} color='text'>
          I‘m a card within a hug!
        </Card>
      </Hug>
    ).toJSON
    expect(json).toMatchSnapshot()
  })

  test('renders text when array of nodes is provided', () => {
    const json = rendererCreateWithTheme(
      <Hug icon={<ThumbsUp />} text={[text, text]}>
        <Card p={3} color='text'>
          I‘m a card within a hug!
        </Card>
      </Hug>
    ).toJSON
    expect(json).toMatchSnapshot()
  })
})
