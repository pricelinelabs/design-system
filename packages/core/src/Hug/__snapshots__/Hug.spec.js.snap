// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hug renders text when array of nodes is provided 1`] = `[Function]`;

exports[`Hug renders text when string is provided 1`] = `[Function]`;

exports[`Hug renders text, icon, and Child 1`] = `[Function]`;

exports[`Hug renders with border-radius from theme on top only 1`] = `
.c2 {
  padding: 8px;
  background-color: #fff;
  color: #001833;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 14px;
}

.c0 {
  border: 1px solid #fff;
  border-radius: 2px;
}

.c1 {
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
  overflow: hidden;
}

.c5 > * {
  border: 0 !important;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
    color="text.lightest"
  >
    <div
      className="c4"
      fontSize={1}
    />
  </div>
  <div
    className="c5"
  />
</div>
`;
